src/framework/step/mod.rs

```diff
@@ -0,0 +1,132 @@
+use crate::framework::Error;
+#[cfg(feature = "enable-serde")]
+use serde::{
+    de::{Error as SerdeError, Visitor},
+    Deserialize, Deserializer, Serialize, Serializer,
+};
+use std::fmt::Formatter;
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Step {
+    Example(ExampleCircuit),
+    ModulusConversion(ModulusConversionDetails),
+}
+
+impl Step {
+    #[must_use]
+    pub fn to_path(&self) -> String {
+        match self {
+            Self::Example(circuit) => format!("example/{}", circuit.to_path()),
+            Self::ModulusConversion(details) => format!("modulus_conversion/{}", details.to_path()),
+        }
+    }
+
+    pub fn from_path(path_str: &str) -> Result<Step, Error> {
```

Comment:
that is probably belong to `TryFrom<&str> for Step` implementation.

Reply:
my only concern with doing this is that it's not particularly descriptive. obviously, we are trying to parse a string into a `Step`, but it's not just "any" string, it's specifically a url path string. what are your thoughts?

Reply:
We can think of this as just a string... That also happens to conform to the stair of a URI path component.